yw_website/yw_website/apps/website/utils.py

```diff
@@ -43,3 +45,56 @@ def search_and_create_query_set(q):
         return Workflow.objects.filter(query).exclude(version__isnull=True)
     else:
         return Workflow.objects.none()
+
+def get_block_data(run_id):
+    '''
+    This function sets up the block structure so that from a parent program
+    block you may drill down into a child of that program block
+    '''
+    program_blocks = ProgramBlock.objects.filter(run=run_id)
+    data = Data.objects.filter(run=run_id)
+    # print(data)
+
+    blocks = []
+    data_vals = []
+
+    # grab parent block
+    parents = get_direct_descendants(None, program_blocks)
+    print(parents)
+    # for data in parent_data:
+    #     parent.data_objs.append(data)
+
+    # grab child blocks
+    child_blocks = program_blocks.exclude(in_program_block_id=None)
+    for block in child_blocks:
+        next_block = ProgramBlock()
+        next_block.name = block.name
+        next_block.programblock_id = block.programblock_id
+        next_block.id = block.id
+        next_block.in_program_block_id = block.in_program_block_id
+        next_block.direct_descendents = program_blocks.filter(in_program_block=next_block.id)
+
+        # block_data = data.filter(in_program_block=next_block.id)
+        # for data in block_data:
+        #     next_block.data_objs.append(data)
+
+        blocks.append(next_block)
```

Comment:
wait remove these blocks references, and you need to return the list of `parents`